### PR TITLE
remove siteBasePath from routes and always prepend to sitePath

### PR DESCRIPTION
**Pull request by 928PJY**
#4222 

require changes of config migration tool @mingwli :
1. remove `siteBasePath` from routes

### CODE REVIEW

File: docs/specs/metadata.yml [hunk 1]
```diff
@@ -95,6 +97,8 @@ inputs:
     product: Win
     files: "**/*"
     baseUrl: https://docs.com/sccm
+    routes:
```
Comment (OsmondJiang) [inline docs/specs/metadata.yml]: there are two concerns:
- User need combine two config "baseUrl" and "routes" to get the finally site url
- Routes expose something like `internal mapping` out to end user, it doesn't map to output path or site path

Comment (OsmondJiang) [reply docs/specs/metadata.yml]: actually the `routes` in this change is more like `sourceBasePath`.

Comment (OsmondJiang) [reply docs/specs/metadata.yml]: we can provide `sourceBasePath` + `baseUrl` experiences to docs user and `routes` to community users

File: src/docfx/docset/Docset.cs [hunk 2]
```diff
@@ -170,11 +170,11 @@ internal class Docset
             Config = config;
             DocsetPath = PathUtility.NormalizeFolder(Path.GetFullPath(docsetPath));
             Locale = locale.ToLowerInvariant();
-            Routes = NormalizeRoutes(config.Routes);
+            (HostName, SiteBasePath) = SplitBaseUrl(config.BaseUrl);
+            Routes = NormalizeRoutes(config.Routes,  config.SourceBasePath, SiteBasePath);
```
Comment (yufeih) [inline src/docfx/docset/Docset.cs]:
```suggestion
            Routes = NormalizeRoutes(config.Routes, config.SourceBasePath, SiteBasePath);
```

File: src/docfx/docset/Docset.cs [hunk 3]
```diff
@@ -224,7 +224,7 @@ Repository GetRepositoryInternal(string fullPath)
             }
         }
 
-        private static IReadOnlyDictionary<string, string> NormalizeRoutes(Dictionary<string, string> routes)
+        private static IReadOnlyDictionary<string, string> NormalizeRoutes(Dictionary<string, string> routes, string sourceBasePath, string siteBasePath)
```
Comment (yufeih) [inline src/docfx/docset/Docset.cs]: where is `sourceBasePath` used?

Comment (928PJY) [reply src/docfx/docset/Docset.cs]: line 239

---
In reply to: [281133469](https://github.com/dotnet/docfx/pull/4481#discussion_r281133469) [](ancestors = 281133469)

File: docs/specs/metadata.yml [hunk 4]
```diff
@@ -199,16 +193,15 @@ inputs:
...
+    sourceBasePath: articles
     baseUrl: https://docs.com/azure
-    routes:
-      articles/: "azure/"
   articles/iot-central/concepts-architecture.md:
   articles/billing/billing-add-change-azure-subscription-administrator.md:
 outputs:
   azure/iot-central/concepts-architecture.json: |
-    {"document_id":"7e13132e-7847-dc4f-90d5-32a233d1c81a","document_version_independent_id":"d9cc5ab8-926a-8a5a-312c-10ca36619bad"}
+    {"document_id":"57901a4c-fd14-9bd5-0dc2-fbd965db5287","document_version_independent_id":"7d7a36da-9cce-c2dd-3e04-ce9ae5e197e5"}
```
Comment (OsmondJiang) [inline docs/specs/metadata.yml]: if the document id changed, means we did something wrong here

File: src/docfx/docset/Docset.cs [hunk 5]
```diff
@@ -225,15 +225,24 @@ Repository GetRepositoryInternal(string fullPath)
             }
         }
 
-        private static IReadOnlyDictionary<string, string> NormalizeRoutes(Dictionary<string, string> routes)
+        private static IReadOnlyDictionary<string, string> NormalizeRoutes(Dictionary<string, string> routes, string sourceBasePath)
```
Comment (yufeih) [inline src/docfx/docset/Docset.cs]: consider changing the logic inside `ApplyRoutes` over here, that is where all the logic is applied, this method just does path normalization.

Comment (928PJY) [reply src/docfx/docset/Docset.cs]: if I move this logic into ApplyRoutes, this logic will run everytime the apply routes is called
this method is to normalize the routes, which is exactly what I am doing, right? 

---
In reply to: [284527991](https://github.com/dotnet/docfx/pull/4481#discussion_r284527991) [](ancestors = 284527991)

**Pull request closed without merging**